crates/sui-types/src/messages_checkpoint.rs

```diff
@@ -399,6 +401,13 @@ pub struct CheckpointSignatureMessage {
 #[derive(Clone, Debug, Serialize, Deserialize, JsonSchema)]
 pub struct CheckpointContents {
     transactions: Vec<ExecutionDigests>,
+    /// This field 'pins' user signatures for the checkpoint:
+    ///
+    /// * For normal checkpoint this field will contain same number of elements as transactions.
+    /// * Genesis checkpoint has transactions but this field is empty.
+    /// * Last checkpoint in the epoch will have (last)extra system transaction
+    /// in the transactions list not covered in the signatures list
+    user_signatures: Vec<Signature>,
```

Comment:
This seems, not so good. Can we instead include the signature inline with the transaction its for?